OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2305,7 +2305,7 @@ public void run() {
       if (shouldLogUserPrivacyConsentErrorMessageForMethodName(OSTaskController.SET_SUBSCRIPTION))
          return;
 
-      getCurrentSubscriptionState(appContext).setUserSubscriptionSetting(disable);
+      getCurrentSubscriptionState(appContext).setPushDisabled(!disable);
       OneSignalStateSynchronizer.setSubscription(disable);
```

Comment:
This line need to be updated to `OneSignalStateSynchronizer.setSubscription(!disable);` or we change the name `disablePush` as well an update the internals to match.

Reply:
Given that subscription can also change email Synchronizer, seems more readable to keep it as setSubscription